### PR TITLE
MC-1566: Retrieve Fakespot JSON from GCS bucket and return in response

### PR DESCRIPTION
**Pull request by katerinachinnappan**
## References

JIRA: https://mozilla-hub.atlassian.net/browse/MC-1566

## Description
Initializing GCS client for Fakespot, retrieving products data from GCS bucket & returning in response. Tracking metrics such as products count, blob size, last updated, etc.

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

### CODE REVIEW

File: merino/configs/default.toml [hunk 1]
```diff
@@ -485,6 +485,23 @@ bucket_name = ""
...
+# The maximum size in bytes of the GCS fakespot products blob. If exceeded, an error will be logged.
+max_size = 2_000_000
+
+# MERINO__CURATED_RECOMMENDATIONS__GCS__FAKESPOT__BLOB_PREFIX
+# GCS path of the fakespot products JSON file.
+blob_name = "products.json"
+
+# MERINO__CURATED_RECOMMENDATIONS__GCS__FAKESPOT__CRON_INTERVAL_SECONDS
+# Interval in seconds at which the cron job checks if GCS contains updated fakespot products data.
+cron_interval_seconds = 3600 # 60 minutes
```
Comment (katerinachinnappan) [inline merino/configs/default.toml:503]: setting to 1 hr as per slack thread: https://mozilla.slack.com/archives/C07P8818LPK/p1729644429006069

File: merino/configs/default.toml [hunk 2]
```diff
@@ -485,6 +485,23 @@ bucket_name = ""
...
 gcp_project = ""
 
+fakespot_bucket_name = ""
+fakespot_gcp_project = ""
+
+
+[default.curated_recommendations.gcs.fakespot]
+# MERINO__CURATED_RECOMMENDATIONS__GCS__FAKESPOT__MAX_SIZE
+# The maximum size in bytes of the GCS fakespot products blob. If exceeded, an error will be logged.
+max_size = 2_000_000
```
Comment (katerinachinnappan) [inline merino/configs/default.toml:495]: setting to this for now.

File: merino/configs/stage.toml [hunk 3]
```diff
@@ -31,3 +31,6 @@ bucket_name = "merino-airflow-data-stagepy"
 # MERINO__CURATED_RECOMMENDATIONS__GCS__GCP_PROJECT
 # GCP project name where the GCS bucket lives.
 gcp_project = "moz-fx-merino-nonprod-ee93"
+
+fakespot_bucket_name = "fakespot-nonprod-stage-newtab-products"
```
Comment (katerinachinnappan) [inline merino/configs/stage.toml:35]: i've tested locally only with prod as I only have access to the prod bucket. can we please confirm these are correct values for staging?

**Review by srbiv — APPROVED** [review]

**Review by jpetto — APPROVED** [review]
this is looking great! we'll ship when you're back on monday or tuesday 🚀

**Pull request merged**